core/src/main/java/tech/tablesaw/io/string/DataFramePrinter.java

```diff
@@ -189,22 +192,25 @@ private String tableName(Relation frame, int width) {
             int i;
             for (i = 0; i < Math.ceil((double) rowCount / 2); i++) {
                 for (int j = 0; j < colCount; j++) {
-                    data[i][j] = frame.getString(i, j);
+                    Column<?> col = frame.column(j);
+                    data[i][j] = (col.size() > i ? col.getString(i) : TOO_SHORT_COLUMN_MARKER);
                 }
             }
             for (int j = 0; j < colCount; j++) {
                 data[i][j] = "...";
             }
             for (++i; i <= rowCount; i++) {
                 for (int j = 0; j < colCount; j++) {
-                    data[i][j] = frame.getString(frame.rowCount() - maxRows + i - 1, j);
+                    Column<?> col = frame.column(j);
+                    data[i][j] = (col.size() > frame.rowCount() - maxRows + i - 1 ? col.getString(frame.rowCount() - maxRows + i - 1) : TOO_SHORT_COLUMN_MARKER);
                 }
             }
         } else {
             data = new String[rowCount][colCount];
             for (int i = 0; i < rowCount; i++) {
                 for (int j = 0; j < colCount; j++) {
-                    String value = frame.getString(i, j);
+                    Column<?> col = frame.column(j);
+                    String value = (col.size() > i ? col.getString(i) : TOO_SHORT_COLUMN_MARKER);
```

Comment:
It might be nice to make a method that refactors out the common logic:

```
private String getString(Column<?> col, index i) {
  return col.size() > i ? col.getString(i) : TOO_SHORT_COLUMN_MARKER;
}
```